src/utils/coingecko_types.ts

```diff
@@ -0,0 +1,31 @@
+export interface CoinGeckoResponse {
+	/**
+	 * Whether the response status code returned a successful code (>200 && <300).
+	 */
+	success: boolean;
+	/**
+	 * The response status message
+	 */
+	message: string;
+	/**
+	 * The response status code
+	 */
+	code: number;
+	/**
+	 * The body data in JSON format from the request.
+	 */
+	data: any;
+}
+
+export interface CoinGeckoPriceRequestParams extends Record<string, string> {
+	// CSV of the coin IDs
+	ids: string;
+	// CSV of the fiat currencies to be quited in
+	vs_currencies: string;
+}
+
+export interface CoinGeckoPriceResponseData {
```

Comment:
Since this only works for requests that were made for AR prices, consider using AR in the interface name, i.e. `CoinGeckoARPriceResponseData`

Reply:
I made that type more generic instead

Reply:
6f3b6aa8f9950684933f276c09c0813081366572